.github/workflows/ci.yml

```diff
@@ -117,14 +117,16 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
fixes strange errors that appear when master is ahead of the current branch. By default this action will merge master with the PR on checkout, which can cause confusion when you encounter error messages related to code changes not on the current branch.